.github/workflows/pr_verify.yaml

```diff
@@ -48,25 +48,40 @@ jobs:
       with:
         go-version: '1.22'
 
+    - name: Get Dagger version to install
+      run: |
+        # convert the string of changed folders into an array
+        mapfile -t CHANGED_MODULES < "changed_modules.txt"
+
+        echo "Changed modules: ${CHANGED_MODULES[@]}"
+
+        if [ ${#CHANGED_MODULES[@]} -eq 0 ]; then
+          echo "No changed folders found."
+          exit 0
+        fi
```

Comment:
I would merge this script with the previous one in get_changed_files, or at least leverage the use of the vars calculated there, for example `if: SKIP_UNIT_TEST == true` should avoid this check.